microsoft/testsuites/cvm/cvm_azure_host.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from pathlib import Path
+from typing import Any, Dict
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+)
+from lisa.testsuite import TestResult
+from lisa.tools import Dmesg
+from lisa.util import LisaException
+
+
+@TestSuiteMetadata(
+    area="cvm",
+    category="functional",
+    description="""
+    This test suite is for azure host vm pre-checks
+    for nested-cvm cases.
+    """,
+)
+class CVMAzureHostTestSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            Runs Dmesg tool to get kernel logs
+            and verify if azure vm is snp enabled.
+        """,
+        priority=3,
+    )
+    def verify_azure_vm_snp_enablement(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        sev_enabled_pattern = r"mshv: SEV-SNP is supported"
```

Comment:
the normal azure vm will fail in this case or not?

Reply:
yes, i guess, it will if we dont find that message. This supposed to run only for nested CVM supported SKU and for Dom0 image only.

Reply:
I can not skip the testcase because it has to be failed if it does not find the message. What we can do it, if OS is Mariner then only we can execute it. Let me know